clients/command/src/main/java/org/eclipse/hono/client/command/CommandRouterCommandConsumerFactory.java

```diff
@@ -288,4 +354,42 @@ private Future<Void> removeCommandConsumer(
                     }
                 });
     }
+
+    private Future<Void> handleClosingConsumers(final SpanContext onReleaseSpanContext) {
+        if (pendingConsumersToClose.size() >= batchSize) {
+            return removeCommandConsumers(pendingConsumersToClose.size(), onReleaseSpanContext);
```

Comment:
How about cancelling an existing timer here? Otherwise that timer execution could cause a `unregisterCommandConsumers` request with only a few items, being sent ahead of `batchMaxTimeout`.

Reply:
It makes sense, will adapt.